### ISSUE TITLE
Add possibility to manually rotate individual tiles

### ISSUE THREAD

**Issue by zifredder** [PreibischLab/BigStitcher#66]
Great plugin, thanks for the work! This is a feature request.

**TL; DR: Could manual X/Y/Z rotation of tiles be implemented?** Like manual X/Y/Z transformation (via _Stitching Explorer > Preprocessing > Arrange Views > Manually translate Views_).

My use case is the following: I have several confocal image stacks taken from the same specimen, each stack shows a different part of the specimen, imaged at different rotations (around the Z-axis) of the specimen. These tiles may have overlapping regions. The rotation of the specimen for each tile is unknown (i. e., there is no metadata information on it). I want to stitch the tiles together.
Before stitching, I need to correct the rotation manually, because the stitching algorithms assume the same Z-axis-orientation of the specimen for all tiles (right?*). It would be great if the tile rotation could be manually arranged for stitching, just like the X/Y/Z position can be manually adjusted.
I feel that this functionality is already very close, because the tiles can be manually transformed, including rotation, in the BigDataViewer: By selecting _Tools > Manual Transform_, each tile can be interactively translated and rotated. However, the transformations done in the BigDataViewer are discarded when the tiles are clicked on in the _Stitching Explorer_. Maybe the transformation data from the BigDataViewer could just be applied and stored as the modified tile coordinates?

*My assumption is that Z-axis rotation is a fixed parameter for the stitching algorithms. I would be happy to learn that there is a way to make the algorithms also try different tile rotations, not only different X/Y/Z transformations.

I hope I have described my feature request clearly. Please let me know if I should elaborate anything more deeply. I can also provide a sample data set if that helps.

Cheers!

**Comment by StephanPreibisch** [comment]
Hi, there are several ways you can do this. Could you send me a minimal example so I can advice best?

**Comment by zifredder** [comment]
Great, thanks for the quick reply and helping me out!

Here are two downsampled stacks of a [locust brain ](https://www.frontiersin.org/files/Articles/1052/fnsys-03-021/image_n/fnsys-03-021-g001.gif)with some stained neurons: https://box.uni-marburg.de/index.php/s/matS3W9r82ZkjBj
I hope the relation of the stacks is clear -- locust_0.tif belongs to the left of locust_1.tif; there is a large and bright soma near the brain midline, visible in both stacks, as they have some overlap.

(Just in case you wonder: I know that the stack rotation is stored in the meta data of these files, so they could be easily rotated to restore the exact orientation. That is usually not the case, I just chose these stacks as an illustrative example.)

Please let me know if something is unclear or anything else is necessary -- thanks a bunch!

**Comment by StephanPreibisch** [comment]
Hi, the rotation between those two images is not obvious to me ... what the approximate angle and axis?

**Comment by zifredder** [comment]
Sorry for not making it clear.
In the [BigDataViewer screenshot](https://user-images.githubusercontent.com/11832636/76453748-4f68ca80-63cb-11ea-9a5e-e6ad7aa63d5b.png) above: The stack arrangement can be seen in the top left corner of the window, where the left square represents locust_0.tif and the right square represents locust_1.tif. The viewing angle is along the z-axis. locust_0.tif is the green stack and locust_1.tif is the magenta stack. The green stack is not translated and not rotated, but the magenta stack is translated to the right (along the x-axis) by about one stack-width, translated up (y-axis) a bit and rotated by ~30 deg counter clock-wise (viewed from this direction) around the z-axis. 
The cell soma that is present in both stacks is the bright white dot in the screenshot center; it is inside the triangle where the stacks overlap (the intersection of the two squares in the top left corner).

Does this explanation make sense? Please tell me if something is still not clear.

**Comment by StephanPreibisch** [comment]
Hi, made a little video and put it on twitter: https://twitter.com/preibischs/status/1238113768623783937?s=20

Please let me know if that fixes your problems. As mentioned, precise alignment would be done using interest points. Let me know if I can help with that too?

**Comment by StephanPreibisch** [comment]
And a YouTube link with higher quality https://youtu.be/TLIGvJYyogI

**Comment by zifredder** [comment]
Terrific! Thanks a lot!
Precise alignment using interest points would be ideal -- if you don't mind, please elaborate.

I already tried around with, but I guess I can't find the right settings (although the problem might be that I am not into image processing at all ...). I don't even know whether this would be done in _Stitching_ or _Multiview_ mode (I suspect _Multiview_).
Interest point detection seems straightforward. I used these settings:
![grafik](https://user-images.githubusercontent.com/11832636/76539134-a2508980-6480-11ea-8bb0-fbda8efc1bed.png)
![grafik](https://user-images.githubusercontent.com/11832636/76539203-bd22fe00-6480-11ea-9e40-b566e81d2b31.png)
And obtained these interest points:
![grafik](https://user-images.githubusercontent.com/11832636/76539352-f3607d80-6480-11ea-841e-c03d79b6b8ac.png)
Assuming this is OK so far, I am stuck at this point. I tried _Register using Interest Points_ and (in _Stitching_ mode) _Sticht dataset (expert) > Interest Point based_, but I can't figure out the parameters.
If I understand the [wiki entry](https://imagej.net/BigStitcher_Registration) correctly, these settings should work:
![grafik](https://user-images.githubusercontent.com/11832636/76540327-4a1a8700-6482-11ea-8fc2-f0ef668acab8.png)
But I am clueless as to which transformation model and regularization model to choose. Am I even on the right track?

Thank you so much for your time!

**Comment by StephanPreibisch** [comment]
Hi, I made another video: https://twitter.com/preibischs/status/1238458770310811648?s=20

Let me know if this helps!

**Comment by zifredder** [comment]
Perfect! This will save my colleagues and me so much time -- thanks a lot!
